components/backup-stream/src/metadata/client.rs

```diff
@@ -698,21 +702,41 @@ impl<Store: MetaStore> MetadataClient<Store> {
         Ok(min_checkpoint)
     }
 
+    fn cached_checkpoint(&self, task: &str) -> Option<Checkpoint> {
+        self.caches
+            .get(task)
+            .and_then(|x| x.value().get())
+            .map(|x| Checkpoint {
+                provider: CheckpointProvider::Global,
+                ts: x,
+            })
+    }
+
+    fn update_cache(&self, task: &str, checkpoint: TimeStamp) {
+        let mut c = self.caches.entry(task.to_owned()).or_default();
+        c.value_mut().update(checkpoint);
+    }
+
     pub async fn get_region_checkpoint(&self, task: &str, region: &Region) -> Result<Checkpoint> {
+        if let Some(c) = self.cached_checkpoint(task) {
```

Comment:
Whether we need to consider that the checkpoint from PD is larger than cached checkpoint in local TiKV ?

Reply:
This is what the "lease" did, the lease time is 12s (which is the same as TiDB's tick interval), the checkpoint probably won't be increased in this interval.

Reply:
(In fact, perhaps we can reduce the lease time in the future)